itegration: husqvarna_automower_ble: Initial commit (#30921)
home-assistant/home-assistant.github.io: source/_integrations/husqvarna_automower_ble.markdown

---
title: Husqvarna Automower BLE
description: Instructions on how to integrate Husqvarna Automower BLE lawn mowers into Home Assistant.
ha_category:
  - Lawn Mower
ha_release: 2024.2
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@alistair23'
ha_platforms:
  - lawn_mower
ha_integration_type: integration
ha_domain: husqvarna_automower_ble
---

> The Husqvarna Automower BLE integration provides connectivity with Husqvarna Automowers lawn mowers via a local Bluetooth connection. This allows connecting and controlling an Automower without any accounts, cloud or network connection.

Reviewer: ```suggestion
The Husqvarna Automower BLE integration provides connectivity with Husqvarna Automowers lawn mowers via a local Bluetooth connection. This allows connecting and controlling an Automower without any accounts, cloud, or network connection.
```

The integration is based on [AutoMower-BLE](https://github.com/alistair23/AutoMower-BLE), a unofficial reverse engineered Husqvarna Automower Connect BLE library.

> ### Tested an verified mowers

Reviewer: ```suggestion
### Tested and verified mowers
```

> This has been tested against the following mowers, although it is expected that all will work

Reviewer: ```suggestion
It is expected that all Husqvarna mowers will work. However, the integration has been tested against the following mowers:
```

>  - Automower 305

Reviewer: ```suggestion

  - Automower 305
```

### Prerequisites

1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/). An ESPHome Bluetooth proxy works well and allows locating a device close to the mower.

> 2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).

Reviewer: ```suggestion
2. Enter the pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
```

---
title: Husqvarna Automower BLE
description: Instructions on how to integrate Husqvarna Automower BLE lawn mowers into Home Assistant.
ha_category:
  - Lawn Mower
ha_release: 2024.2
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@alistair23'
ha_platforms:
  - lawn_mower
ha_integration_type: integration
ha_domain: husqvarna_automower_ble
---

The Husqvarna Automower BLE integration provides connectivity with Husqvarna Automowers lawn mowers via a local Bluetooth connection. This allows connecting and controlling an Automower without any accounts, cloud, or network connection.

> The integration is based on [AutoMower-BLE](https://github.com/alistair23/AutoMower-BLE), an unofficial reverse engineered Husqvarna Automower Connect BLE library.

Reviewer: ```suggestion
The integration is based on [AutoMower-BLE](https://github.com/alistair23/AutoMower-BLE), an unofficial reverse engineered Husqvarna Automower Connect BLE library.

There is currently support for the following device types within Home Assistant:

- Lawn Mower
- Battery Level
```

### Tested and verified mowers

It is expected that all Husqvarna mowers will work. However, the integration has been tested against the following mowers:

  - Automower 305

### Prerequisites

1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/). An ESPHome Bluetooth proxy works well and allows locating a device close to the mower.
2. Enter the pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
3. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.

Pairing can take a few goes. Even when using the official Android application, it can be tricky to get the first pair to succeed. If you are having issues, reboot the mower and try again.

{% include integrations/config_flow.md %}

> There is currently support for the following device types within Home Assistant:
> 
> - Lawn Mower
> - Battery Level

Reviewer: ```suggestion
```

  - lawn_mower
ha_integration_type: integration
ha_domain: husqvarna_automower_ble
---

The Husqvarna Automower BLE integration provides connectivity with Husqvarna Automowers lawn mowers via a local Bluetooth connection. This allows connecting and controlling an Automower without any accounts, cloud or network connection.

The integration is based on [AutoMower-BLE](https://github.com/alistair23/AutoMower-BLE), a unofficial reverse engineered Husqvarna Automower Connect BLE library.

### Tested an verified mowers

This has been tested against the following mowers, although it is expected that all will work
 - Automower 305

### Prerequisites

1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/). An ESPHome Bluetooth proxy works well and allows locating a device close to the mower.

2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).

> 4. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.

Reviewer: ```suggestion
3. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.
```

> 1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/). An ESPHome Bluetooth proxy works well and allows locating a device close to the mower.
> 
> 2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
> 
> 3. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.

Reviewer: ```suggestion
1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/). An ESPHome Bluetooth proxy works well and allows locating a device close to the mower.
2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
3. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.
```

---
title: Husqvarna Automower BLE
description: Instructions on how to integrate Husqvarna Automower BLE lawn mowers into Home Assistant.
ha_category:
  - Lawn Mower
ha_release: 2024.2
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@alistair23'
ha_platforms:
  - lawn_mower
ha_integration_type: integration
ha_domain: husqvarna_automower_ble
---

The Husqvarna Automower BLE integration provides connectivity with Husqvarna Automowers lawn mowers via a local Bluetooth connection. This allows connecting and controlling an Automower without any accounts, cloud or network connection.

The integration is based on [AutoMower-BLE](https://github.com/alistair23/AutoMower-BLE), a unofficial reverse engineered Husqvarna Automower Connect BLE library.

### Tested an verified mowers

This has been tested against the following mowers, although it is expected that all will work
 - Automower 305

> ### Home Assistant

Reviewer: ```suggestion
### Prerequisites
```

> 1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/), an ESPHome Bluetooth proxy works well and allows locating a device close to the mower.

Reviewer: ```suggestion
1. Setup a [Bluetooth controller](https://www.home-assistant.io/integrations/bluetooth/). An ESPHome Bluetooth proxy works well and allows locating a device close to the mower.
```

> 2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305 for example the mower will enter pairing mode for the first 3 minutes after power on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).

Reviewer: ```suggestion
2. Enter pairing mode on the mower. Different models will do this in different ways. For the 305, for example, the mower will enter pairing mode for the first 3 minutes after powering on. Ensure the mower is in pairing mode when adding the integration. This only needs to be done once per BLE controller (so changing the ESPHome device will require a repair).
```

> 3. Add the integration to your Home Assistant installation:
>    [![my_button](https://my.home-assistant.io/badges/config_flow_start.svg)](https://my.home-assistant.io/redirect/config_flow_start/?domain=husqvarna_automower_ble)
> 

Reviewer: ```suggestion
```
Removing this, as it is part of the config_flow

> 4. When prompted you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, an Android phone or some other means

Reviewer: ```suggestion
4. When adding the integration to Home Assistant, you will need to enter the mower BLE Mac address. You can find this in the ESPHome logs, on an Android phone, or by some other means.
```

> 
> 5. Confirm successful connection of mower and assign to an area.

Reviewer: ```suggestion
```

> Pairing can take a few goes. Even when using the official Android application it can be tricky to get the first pair to succeed. If you are having issues reboot the mower and try again.

Reviewer: ```suggestion
Pairing can take a few goes. Even when using the official Android application, it can be tricky to get the first pair to succeed. If you are having issues, reboot the mower and try again.
```
